samples/cellular/modem_shell/src/sock/sock_shell.c

```diff
@@ -100,8 +96,29 @@ static const char sock_usage_str[] =
 	"                            Examples of equivalent hexadecimal data strings:\n"
 	"                                010203040506070809101112\n"
 	"                                01 02 03 04 05 06 07 08 09 10 11 12\n"
-	"                                01020304 05060708 09101112\n"
-	"Options for 'rai' command:\n"
+	"                                01020304 05060708 09101112";
+
+static const char sock_recv_usage_str[] =
+	"Usage: sock recv [options]\n"
+	"Options:\n"
+	"  -i, --id, [int]           Socket id. Use 'sock list' command to see open\n"
+	"                            sockets.\n"
+	"  -r, --start,              Initialize variables for receive throughput\n"
+	"                            calculation\n"
+	"  -l, --length, [int]       Length of expected data in bytes. After receiving\n"
+	"                            data with given length, summary of data throughput\n"
+	"                            is printed. Should be used with -r option.\n"
+	"  -B, --blocking, [int]     Blocking (1) or non-blocking (0) mode. This only\n"
+	"                            accounts when -r is given. Default value is 0.\n"
+	"  -P, --print_format, [str] Set receive data print format: 'str' (default) or\n"
+	"                            'hex'";
+
+static const char sock_rai_usage_str[] =
+	"Usage: sock rai -i <socket id> [--rai_last] [--rai_no_data] [--rai_one_resp]\n"
+	"       [--rai_ongoing] [--rai_wait_more]\n"
```

Comment:
Because only one RAI option can be given at a time, maybe the syntax should be like this:
```suggestion
	"Usage: sock rai -i <socket id> [--rai_last | --rai_no_data | --rai_one_resp |\n"
	"       --rai_ongoing | --rai_wait_more]\n"
```
At least similar syntax has been used for `sock send`.

Reply:
Hmmm, the code allows to give one or many of these options.

Reply:
That should probably be corrected. The socket uses the last value set for the next send call, except no_data which gets applied immediately.

Reply:
Jira ticket created for this: MOSH-527